fem/fe/fe_base.cpp

```diff
@@ -2521,6 +2530,116 @@ void NodalTensorFiniteElement::SetMapType(const int map_type)
    }
 }
 
+const DofToQuad &NodalTensorFiniteElement::GetDofToQuad(
+   const IntegrationRule &ir,
+   DofToQuad::Mode mode,
+   const ElementDofOrdering ordering) const
+{
+   MFEM_VERIFY(!(mode == DofToQuad::Mode::TENSOR &&
+                 ordering == ElementDofOrdering::NATIVE),
+               "Invalide combination of DofToQuad::Mode and ElementDofOrdering.");
```

Comment:
@YohannDudouit @v-dobrev I finshed the `FULL` and `LEXICOGRAPHIC` case for the different element and derivative types for `NodalTensorFiniteElement`. The combination of `TENSOR` and `NATIVE` doesn't make sense so I have a check against that. This function has the logic structure as the original `FiniteElement::DofToQuad` but it simply undoes the native ordering using `dof_map`. I had to move `ElementDofOrding` from `fespace.hpp` to `fe_base.hpp` since it's an enum, but I think it still makes sense since it's an element level concept.

I'm done making changes to the PR for now. Let me know what you think.